api/src/main/java/graphql/nadel/instrumentation/ChainedNadelEngineInstrumentation.java

```diff
@@ -14,14 +11,13 @@ public ChainedNadelEngineInstrumentation(List<NadelInstrumentation> instrumentat
     }
 
     @Override
-    public RootExecutionResultNode instrumentRootExecutionResult(RootExecutionResultNode rootExecutionResultNode, NadelInstrumentRootExecutionResultParameters parameters) {
+    public void instrumentRootExecutionResult(Object rootExecutionResult, NadelInstrumentRootExecutionResultParameters parameters) {
```

Comment:
`Object rootExecutionResult` is actually `Either<RootExecutionResultNode, ExecutionResult>` (left - current gen, right - nextgen)

this instrumentation method is used by gateway's tracer library.
to replicate the functionality that the tracer library has around `RootExecutionResultNode` argument, just supplying `ExecutionResult` is not going to be enough I believe. For example, tracer uses `RootExecutionResultNode`'s `ElapsedTime elapsedTime` field which does not exist in `ExecutionResult`. 
So we either need to extend `ExecutionResult` and add more stuff to it or wrap it around with timings and other stuff and pass that wrapped class to `instrumentRootExecutionResult`

Reply:
I don't think we need this, I think `beginExecute` already provides this value in the `onComplete`? From the `NadelInstrumentationExecuteOperationParameters` we even have the `ExecutableNormalizedOperation`.

Also just ranting below, not really necessary to read.

I'm always weary of having `Object` as the type. It's not a big deal to add another function with the new type, but when you change something to `Object` it's always a PITA because you or other devs never know what the value _could_ be unless you read documentation, see working with `ResultPath#getSegmentValue` or `NormalizedInputValue#getValue`. Little to no documentation about what the `Object` actually can be, makes it very error prone. There's next to 0 type safety around this.

In the ideal world we'd be using union types. The closest thing we have to that in Kotlin is sealed types to wrap the original objects e.g.

```kotlin
sealed class JsonNodePathSegment<T : Any> {
    abstract val value: T

    data class Int(override val value: kotlin.Int) : JsonNodePathSegment<kotlin.Int>()
    data class String(override val value: kotlin.String) : JsonNodePathSegment<kotlin.String>()
}
```